src/cli/self_update.rs

```diff
@@ -378,8 +385,8 @@ fn check_existence_of_rustc_or_cargo_in_path(no_prompt: bool) -> Result<()> {
     // Only the test runner should set this
     let skip_check = env::var_os("RUSTUP_INIT_SKIP_PATH_CHECK");
 
-    // Ignore this check if called with no prompt (-y) or if the environment variable is set
-    if no_prompt || skip_check == Some("yes".into()) {
+    // Skip this if the environment variable is set
+    if skip_check == Some("yes".into()) {
```

Comment:
Why hav eyou ceased obeying `no_prompt` here?

Reply:
I changed it to show output even when `no_prompt` is used, but still continue afterwards.

Reply:
Aah yes, I see.  I shall have to give that a go and decide if I like the feel of it, but not until the other issue is resolved and you've rebased things.  It'll be a final check that I'm OK with the feel of the whole feature.

I wonder if we need the two environment variables documented in `README.md`?  Perhaps so, though perhaps as a separate PR afterwards.